src/plugins/unified_histogram/public/container/hooks/use_state_props.ts

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { DataViewField, DataViewType } from '@kbn/data-views-plugin/common';
+import { getAggregateQueryMode, isOfAggregateQueryType } from '@kbn/es-query';
+import { useCallback, useEffect, useMemo } from 'react';
+import { UnifiedHistogramChartLoadEvent, UnifiedHistogramFetchStatus } from '../../types';
+import type {
+  UnifiedHistogramState,
+  UnifiedHistogramStateService,
+} from '../services/state_service';
+
+export const useStateProps = ({
+  state,
+  stateService,
+}: {
+  state: UnifiedHistogramState | undefined;
+  stateService: UnifiedHistogramStateService | undefined;
+}) => {
+  /**
+   * Contexts
+   */
+
+  const isPlainRecord = useMemo(() => {
```

Comment:
Logic for `isPlainRecord` and `isTimeBased` copied from Discover so consumers don't need to reimplement themselves.